TST: fix `to` issue for 8-bit model

yao-matrix opened the pull request:
## Issue
4 cases fail 
```
pytest -rA tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit
pytest -rA tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit_dora
pytest -rA tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit_iter_5
pytest -rA tests/test_gpu_examples.py::TestLoftQ::test_t5_loftq_8bit
```
w/ log

> >               raise ValueError(
>                     "`.to` is not supported for `8-bit` bitsandbytes models. Please use the model as it is, since the"
>                     " model has already been set to the correct devices and casted to the correct `dtype`."
> E                   ValueError: `.to` is not supported for `8-bit` bitsandbytes models. Please use the model as it is, since the model has already been set to the correct devices and casted to the correct `dtype`.

## Root Cause
8-bit models don't support `to` and `cuda` according [transformers logic](https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L4353)

## Fix
check whether model is loaded in 8 bit, and skip `to` if so.

## Results:
Pass

@BenjaminBossan , pls help review, thx very much.

githubnemo commented:
Thanks for the suggestion! I wonder if we should skip the test instead since we're "requesting" to use `self.device` but that model doesn't support using that device. WDYT?

yao-matrix commented on src/peft/utils/integrations.py:
```diff
@@ -101,6 +104,7 @@ def dequantize_bnb_weight(weight: torch.nn.Parameter, state=None):
             weight = weight.to(torch.device("cuda"))
         elif is_xpu_available():
             weight = weight.to(torch.device("xpu"))
+            state.SCB = state.SCB.to(torch.device("xpu"))
```
`weight` is moved to accelerator, but `SCB` not, will make later `int8_vectorwise_dequant` cry to say `weight in xpu, but SCB in cpu` and make CPU CI, fail, so move it to accelerator too.

yao-matrix commented on src/peft/utils/integrations.py:
```diff
@@ -121,6 +122,7 @@ def dequantize_bnb_weight(weight: torch.nn.Parameter, state=None):
 
     if is_cpu:
         dequantized = dequantized.to(device)
+        state.SCB = state.SCB.to(device)
```
both should be move back

yao-matrix commented:
> Thanks for the suggestion! I wonder if we should skip the test instead since we're "requesting" to use `self.device` but that model doesn't support using that device. WDYT?

@githubnemo, I cleaned the test cases and the device cases passed, we have `torch_device`, `device` of class `TestLoftQ`, `device` through method arguments, kind of make the test case confused, so I cleaned it, then accelerator(xpu, cuda) case can pass.

for `cpu` failures, its because incomplete tensor transfer btw cpu and accelerator, i commented inline.

Thx very much for review and sorry for before's lazy workaround.

BenjaminBossan reviewed (commented):
Thanks for working on this with 8bit bnb and device placement. Unfortunately, when I run this locally, I still get errors:

```
$ CUDA_VISIBLE_DEVICES=0 pytest tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit_dora tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit_iter_5 tests/test_gpu_examples.py::TestLoftQ::test_t5_loftq_8bit

FAILED tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit[cpu] - assert tensor(nan, grad_fn=<MeanBackward0>) > 0.0
FAILED tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit_dora[cpu] - RuntimeError: invalid argument to getCurrentStream
FAILED tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit_iter_5[cuda] - torch.AcceleratorError: CUDA error: an illegal memory access was encountered
FAILED tests/test_gpu_examples.py::TestLoftQ::test_bloomz_loftq_8bit_iter_5[cpu] - torch.AcceleratorError: CUDA error: an illegal memory access was encountered
FAILED tests/test_gpu_examples.py::TestLoftQ::test_t5_loftq_8bit[cuda] - torch.AcceleratorError: CUDA error: an illegal memory access was encountered
FAILED tests/test_gpu_examples.py::TestLoftQ::test_t5_loftq_8bit[cpu] - torch.AcceleratorError: CUDA error: an illegal memory access was encountered
```

I cannot test on XPU, but for CPU and CUDA, I get these errors. Do these tests pass for you?

Furthermore, I was wondering why these tests are not running on CI and I think it's a simple oversight, there is no test marker for `TestLoftQ`. Could you please add the `@pytest.mark.single_gpu_tests` decorator?
